extensions/wellinks/actions/checkFlourishCustomer/checkFlourishCustomer.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  type Action,
+  type DataPointDefinition,
+  type OnErrorCallback,
+  Category,
+  validate,
+} from '@awell-health/extensions-core'
+import { type settings } from '../../settings'
+import { WellinksFlourishClient } from '../../wellinksFlourishClient'
+import { fromZodError, type ValidationError } from 'zod-validation-error'
+import {
+  FieldsValidationSchema,
+  SettingsValidationSchema,
+  fields,
+} from './config'
+import { z, ZodError } from 'zod'
+
+const dataPoints = {
+  userExists: {
+    key: 'userExists',
+    valueType: 'boolean',
+  },
+} satisfies Record<string, DataPointDefinition>
+
+export const checkFlourishCustomer: Action<
+  typeof fields,
+  typeof settings,
+  keyof typeof dataPoints
+> = {
+  key: 'checkFlourishCustomer',
+  category: Category.WORKFLOW,
+  title: 'Check for Flourish Customer',
+  description: 'Checks Flourish for a customer with the given identifier.',
+  fields,
+  dataPoints,
+  previewable: true,
+  onActivityCreated: async (payload, onComplete, onError): Promise<void> => {
+    try {
+      const {
+        fields: { identifier },
+        settings: { flourishApiKey, flourishApiUrl, flourishClientExtId },
+      } = validate({
+        schema: z.object({
+          fields: FieldsValidationSchema,
+          settings: SettingsValidationSchema,
+        }),
+        payload,
+      })
+
+      const client = new WellinksFlourishClient(
+        flourishApiUrl,
+        flourishApiKey,
+        flourishClientExtId
+      )
+
+      const result = await client.user.exists(identifier)
+      await onComplete({
+        data_points: {
+          userExists: result.toString(),
+        },
+      })
+    } catch (err) {
+      if (err instanceof ZodError) {
```

Comment:
We already handle zod errors in our server --> no need for you to catch them unless you want to display something other than information about the incorrect validation

Reply:
i suggest you remove the entire try/catch mechanism here, given we will already know which activity throws the error.